pkg/fieldowner/field_owner.go

```diff
@@ -0,0 +1,46 @@
+package fieldowner
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Wrap wraps a Client with one that sets the field owner.
+func Wrap(upstream client.Client, owner string) client.Client {
+	return &clientWithOwner{Client: upstream, owner: client.FieldOwner(owner)}
+}
+
+type clientWithOwner struct {
+	client.Client
```

Comment:
I'd prefer not to embedd the Client, because should we extend the Client in the future with a method that needs wrapping, the `clientWithOwner` won't do that but the code will still compile.